llama.h

```diff
@@ -20,7 +20,7 @@
 #endif
 
 #define LLAMA_FILE_VERSION 1
-#define LLAMA_FILE_MAGIC 0x67676d66 // 'ggmf' in hex
```

Comment:
Nit: why change the magic rather than the version? I assumed the plan was to keep the magic constant forever. If you bump the version instead, old executables will recognize new model files and [give a more useful error message](https://github.com/ggerganov/llama.cpp/blob/9cbc404ba6699a9ba4925ea25a60552b13491c7a/llama.cpp#L336). And it's nice to distinguish between "this is definitely model file for this project, but it's the wrong version" vs "this is some random junk we don't know anything about".

(This PR is a very neat bit of engineering; please don't let my nitpick distract from that.)

Reply:
not a nitpick but a real change request :)